Further fixes to Prometheus blackbox exporter endpoints

MoteHue opened the pull request:
Enabled items under prometheus_blackbox_exporter_endpoints_kayobe were not picked up because the attribute gets set to `enabled: 'True'`, rather the than `enabled: true` that the default values get. Likely a similar issue to this:
https://github.com/ansible/ansible/issues/11905#issuecomment-130496173

Alex-Welsh reviewed (commented):
It should be using the [true jinja test](https://jinja.palletsprojects.com/en/3.0.x/templates/#jinja-tests.true) rather than matching against the string, so I'd assume it would accept any truthy value. Evidently not

markgoddard commented on etc/kayobe/kolla/inventory/group_vars/prometheus-blackbox-exporter:
```diff
@@ -9,7 +9,7 @@
   {{ (prometheus_blackbox_exporter_endpoints_kayobe | default([]) +
   prometheus_blackbox_exporter_endpoints_default) |
-  selectattr('enabled', 'true') |
+  selectattr('enabled', 'match', 'true|True') |
```
Would this ignore any that use a proper boolean true, or "yes"?

MoteHue replied:
Rewritten this so it now captures all truthy values correctly. It follows the same format/convention @Alex-Welsh has in his firewalld config PR

MoteHue committed:
Fix prometheus_blackbox_exporter_endpoints to allow truthy values

Alex-Welsh commented on etc/kayobe/pulp.yml:
```diff
@@ -532,6 +532,8 @@ stackhpc_pulp_images_kolla:
   - prometheus-v2-server
   - rabbitmq
+  - rabbitmq-3-12
+  - rabbitmq-3-13
```
Stray config?

MoteHue replied:
Yep, it's gone

Alex-Welsh reviewed (commented):
LGTM but I helped write it so someone else should approve

MoteHue committed:
Add bool to rabbitmq endpoint

MoteHue committed:
Correct horizon_public_endpoint

horizon_public_port is dependent on backend_tls being enabled, rather
than external tls.

Alex-Welsh reviewed (commented):
Slight scope creep but I'll allow it. LGTM but someone else should approve

MoteHue commented:
> Slight scope creep but I'll allow it.

I've kept the changes in separate commits, so at least it will be distinct in the blame.

markgoddard commented on etc/kayobe/kolla/inventory/group_vars/prometheus-blackbox-exporter:
```diff
@@ -6,13 +6,16 @@
+  {% endfor %}
+  {% endif %}
+  {% endfor %}
+  {{ (endpoints + prometheus_blackbox_exporter_endpoints_custom) | unique | list }}
```
There is a `select` in the original filter pipeline. Is it possible that an endpoint could be set to an empty string?

Alex-Welsh replied:
I think if `kolla_same_external_internal_vip` is true, then there will be a lot of empty string endpoints, so select would probably be quite important

MoteHue replied:
Done

MoteHue committed:
Add select
